src/main/java/org/wildfly/security/auth/callback/AuthenticationConfigurationCallback.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.callback;
+
+import javax.security.auth.callback.Callback;
+
+import org.wildfly.security.auth.client.AuthenticationConfiguration;
+
+/**
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class AuthenticationConfigurationCallback implements Callback {
+
+    private final AuthenticationConfiguration authenticationConfiguration;
+
+    public AuthenticationConfigurationCallback(AuthenticationConfiguration authenticationConfiguration) {
```

Comment:
This is backwards to me.  We already have access to authentication configuration via AuthenticationContext - that's what it is for.